launch/launch/some_substitutions_type.py

```diff
@@ -32,3 +34,21 @@
     Substitution,
     collections.abc.Iterable,
 )
+
+
+def is_some_substitutions_type(input: Optional[Union[Any, Iterable[Any]]]) -> bool:
```

Comment:
This looks very similar to [`is_substitution`](https://github.com/ros2/launch/blob/76ade26d3f7bcb6a5a1b72a512a8c098e6d53f04/launch/launch/utilities/type_utils.py#L346). Can we just use `is_substitution` instead?

If not, then I think it would be better to move this function into `type_utils.py`.